app/controllers/report_controller/widgets.rb

```diff
@@ -655,7 +655,17 @@ def widget_set_record_vars(widget)
     #      end
     #      widget.miq_widget_shortcuts = ws
     #    end
-    widget.options[:url] = @edit[:new][:url] unless @edit[:new][:url].blank?
+    if @sb[:wtype] == "rf"
+      if @edit[:new][:feed_type] == "internal"
+        widget.resource = RssFeed.find(@edit[:new][:rss_feed_id]) if @edit[:new][:rss_feed_id]
+        widget.options.delete(:url)
+      else
+        widget.resource = nil
+        widget.options[:url] = @edit[:new][:url]
+      end
+    else
+      widget.resource = @edit[:rpt]
```

Comment:
@isimluk I would suggest you try it and if it looks it's not that simple, we can fix it surgically here and then move it to the model after the fix.  My suggestion about the model is because it's generally easier to test model methods.